models/models.js

```diff
@@ -0,0 +1,29 @@
+const connection = require("../db/connection");
+
+
+exports.fetchTopics = () => {
+    return connection.query("SELECT * FROM topics;")
+        .then((result) => {
+            return result.rows;  
+});
+};
+
+
+exports.fetchArticleById = ({ article_id }) => {
+    if (isNaN(+article_id)) {
+      return Promise.reject({ status: 400, msg: "The article_id has to be a number" });
+    }
+    return connection.query(`SELECT * FROM articles 
+      WHERE article_id = $1;`,
+        [+article_id]
+      )
+      .then((result) => {
+        if (result.rows.length) {
+          return result.rows[0];
+        }
+        return Promise.reject({ status: 404, msg: "The article not found" });
+      })
+      .catch((err) => {
```

Comment:
No catch needed here because it already exists in your controller, you also don't need to promise.reject here - express will handle it all for you